packages/workbox-window/Workbox.mjs

```diff
@@ -0,0 +1,417 @@
+/*
+  Copyright 2019 Google LLC
+
+  Use of this source code is governed by an MIT-style
+  license that can be found in the LICENSE file or at
+  https://opensource.org/licenses/MIT.
+*/
+
+import {EventTargetShim} from './utils/EventTargetShim.mjs';
+import {logger} from './utils/logger.mjs';
+import './_version.mjs';
+
+
+const urlsMatch = (url1, url2) => {
+  return new URL(url1, location).href === new URL(url2, location).href;
+};
+
+
+/**
+ * A class to aid in handling service worker registration, updates, and
+ * reacting to service worker lifecycle events.
+ *
+
+ */
+export class Workbox extends EventTargetShim {
+  /**
+   * Creates a new Workbox instance with a script URL and service worker
+   * options. The script URL and options are the same as those used when
+   * calling `navigator.serviceWorker.register(scriptUrl, options)`. See:
+   * https://developer.mozilla.org/en-US/docs/Web/API/ServiceWorkerContainer/register
+   *
+   * @param {string} scriptUrl The service worker script associated with this
+   *     instance.
+   * @param {Object} options The service worker options associated with this
+   *     instance.
+   */
+  constructor(scriptUrl, options = {}) {
+    super();
+
+    this._scriptUrl = scriptUrl;
+    this._swOptions = options;
+
+    // this._registration = null;
```

Comment:
Any reason to leave these here, commented out? If so, can you put in a `TODO:` or something similar for context?

Reply:
I was using it as a way to keep all instance variables visible from the constructor (closure compiler actually requires this, which is where I got it from, but uglify doesn't remove them, so I commented them out).

Reply:
I added a comment to make this clear. I can remove it later though if we don't like it.